langchain/src/chat_models/googlevertexai.ts

```diff
@@ -0,0 +1,300 @@
+  async _generate(
+    messages: BaseChatMessage[],
+    options: this["ParsedCallOptions"]
+    // runManager is omitted since we can't issue token updates
+  ): Promise<ChatResult> {
+    const instance: GoogleVertexAIChatInstance = this.generateInstance(
+      messages,
+      options
+    );
+
+    const parameters: GoogleVertexAIModelParams = {
+      temperature: this.temperature,
+      topK: this.topK,
+      topP: this.topP,
+      maxOutputTokens: this.maxOutputTokens,
+    };
+
+    const result = await this.connection.request(
+      [instance],
+      parameters,
+      options
+    );
+
+    return this.convertResult(result);
+  }
+
+  _llmType(): string {
+    return "googleVertexAI";
+  }
+
+  generateInstance(
+    messages: BaseChatMessage[],
+    options: this["ParsedCallOptions"]
+  ): GoogleVertexAIChatInstance {
+    // Build the instances in the requeswt, which may be built from
+    // a combination (in highest to lowest priority) of the messages
+    // passed in, the options passed in, and the configuration parameters
+    // passed to the constructor.
+    //
+    // Note that || is used and *not* ?? because we want the empty string
+    // and the empty array to be replaced by a string or array that has
+    // something in it if available. || will do this, but ?? won't.
+    const fromMessages = this.convertMessages(messages);
+    const fromOptions = this.convertOptions(options);
+
+    const context =
+      fromMessages.context || fromOptions.context || this.context || "";
+
+    const convertedExamples = this.convertExamples(this.examples);
+    let examples: GoogleVertexAIChatExample[];
+    if (fromOptions?.examples?.length) {
+      examples = fromOptions.examples;
+    } else if (convertedExamples?.length) {
+      examples = convertedExamples;
+    } else {
+      examples = [];
+    }
+
+    const instance: GoogleVertexAIChatInstance = {
+      context,
+      examples,
+      messages: fromMessages.messages,
+    };
+    return instance;
+  }
+
+  convertMessages(baseMessages: BaseChatMessage[]): GoogleVertexAIChatInstance {
+    let context = "";
+    const messages: GoogleVertexAIChatMessage[] = [];
+
+    baseMessages.forEach((baseMessage) => {
+      if (this.SystemMessage.typeEquals(baseMessage)) {
+        // System messages should be added to the context prompt
+        context += baseMessage.text;
+      } else {
+        // Convert the message and add it to the list of messages
+        // if it is a valid author type.
+        const message = this.convertMessage(baseMessage);
+        if (message.author) {
+          messages.push(message);
+        }
+      }
+    });
+
+    return {
+      context,
+      messages,
+    };
+  }
+
+  convertMessage(baseMessage: BaseChatMessage): GoogleVertexAIChatMessage {
+    const ret: GoogleVertexAIChatMessage = {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      author: this.roleAlias[baseMessage._getType()],
```

Comment:
I'd rather hard-code an alias for now to limit the number of specific abstractions we're introducing here - `BaseChatMessages` have a `name` field that could override this in the future.